streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -58,8 +60,21 @@ public void addChild(final ProcessorNode<Void, Void, ?, ?> child) {
     public void init(final InternalProcessorContext<Void, Void> context) {
         super.init(context);
         this.context = context;
-        keySerializer = prepareKeySerializer(keySerializer, context, this.name());
-        valSerializer = prepareValueSerializer(valSerializer, context, this.name());
+        try {
+            keySerializer = prepareKeySerializer(keySerializer, context, this.name());
```

Comment:
I did dig into `prepareKeySerializer` and `prepareValueSerializer` which both use `WrappingNullableUtils#prepareSerializer()` which might call both `context.keySerde()` and `context.valueSerde()`, and thus, I believe we could currently get an exception when trying to get the key serde, even if default key serde is set, but default value serde is not set?

I think this internal helper method needs some updated, too.